tests/suites/test_suite_bignum_core.misc.data

```diff
@@ -155,6 +155,9 @@ mpi_core_bitlen:"10":5
 Test mbedtls_mpi_core_bitlen 0x0a
 mpi_core_bitlen:"a":4
 
+Test mbedtls_mpi_core_bitlen: 0 limbs
```

Comment:
Core functions don't support 0-limb bignums (see the top of `bignum_core.h`). (Legacy bignum functions do, it's up to them to do the right thing without calling core functions.) So we shouldn't have such a test case. It might happen to work for this particular function, but we wouldn't be bothered if it doesn't.

Most core functions fall into one of two categories: they do some kind of loop over the limb count and a 0-limb array just works, or they do some kind of memcpy over the limbs and a 0-limb array technically has undefined behavior which is caught by UBSan but tends to work when not building with UBSan.

The current implementation of `core_bitlen` has an explicit check `limbs == 0`. I think it's a leftover from the very early days of the core module, before we'd decided that it didn't need to support 0-limb. So this check could be removed, which would save a couple of words of code size.

Reply:
OK, thanks, I was unsure on this point. I've removed the test changes.

Reply:
Before we forget: https://github.com/Mbed-TLS/mbedtls/pull/7491